engine/src/cmd/missile.cpp

```diff
@@ -0,0 +1,364 @@
+                              float difficulty,
+                              const Matrix &transmat,
+                              const Vector &CumulativeVelocity,
+                              bool ResolveLast,
+                              UnitCollection *uc )
+{
+    // First we move the missile by calling the super
+    GameUnit<DummyUnit>::UpdatePhysics2( trans, old_physical_state, accel, difficulty, transmat, CumulativeVelocity, ResolveLast, uc );
+
+
+    // Get the target
+    Unit *target = Target();
+
+    // We have a valid target - we mark this as it affects fuel and TTL
+    if(target != nullptr)
+    {
+        had_target = true;
+    }
+
+    // Try and break lock with ECM
+    target = breakECMLock(target);
+
+    // We have a valid target
+    if(target != nullptr)
+    {
+        // But the target is dead already
+        if (target->hull < 0) {
+            target = nullptr;
+        } else {
+            // What does this do?
+            static unsigned int pointdef = ROLES::getRole( "POINTDEF" );
+            target->graphicOptions.missilelock = true;
+            un_iter i = target->getSubUnits();
+
+            Unit   *su;
+            for (; (su = *i) != nullptr; ++i)
+            {
+                if (su->attackPreference() == pointdef)
+                {
+                    if (su->Target() == nullptr)
+                    {
+                        float speed, range, mrange;
+                        su->getAverageGunSpeed( speed, range, mrange );
+                        if ( ( Position()-su->Position() ).MagnitudeSquared() < range*range )
+                        {
+                            su->Target( this );
+                            su->TargetTurret( this );
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // Go wild
+    // I'm pretty sure this code does nothing so I'm commenting it out
+    /*if (retarget == -1)
+    {
+        if (target)
+            retarget = 1;
+        else
+            retarget = 0;
+    }
+
+    if (retarget && target == nullptr)
+    {
+        Target( nullptr );
+    }*/
+
+    if (proximityFuse(target))
+    {
+        return;
+    }
+
+    if (useFuel(target, had_target))
+    {
+        return;
+    }
+}
+
+Unit* Missile::breakECMLock(Unit* target)
+{
+    if(target == nullptr)
+    {
+        return nullptr;
+    }
+
+    // There are two separate checks for ECM breaking lock here
+    // TODO: pick one
+    float r = rand();
+    float rand_max = static_cast<float>(RAND_MAX);
+    float ecm_value = static_cast<float>(UnitUtil::getECM(target)) * SIMULATION_ATOM/32768; // should it be simulation_atom_var?
+
+    if (r/rand_max < ecm_value)
+    {
+        return nullptr;
+    }
+
+    // Second check
+    static size_t max_ecm = static_cast<size_t>(GameConfig::GetVariable( "physics", "max_ecm", 4));
+    size_t missile_hash = reinterpret_cast<size_t>(this)/16383;
```

Comment:
shouldn't this be 16384? (16k)? Line 287 is 32768 (32k)

Reply:
It might, but these two are only somewhat related. One (32k) generates the ECM value. The other (16k) generates a random value to be compared against the ecm value. Not the way I would have implemented it. I vote we leave it as it was.

Reply:
okay - just wanting to verify as the value was off-by one for a true 16k value.